site/src/components/form.md

```diff
@@ -0,0 +1,176 @@
+			class="input"
+			type="text"
+			placeholder="e.g., Why is the sky blue?"
+		/>
+	</div>
+</div>
+<div class="field">
+	<div class="field-section">
+		<label class="field-section-label label" for="sample-form-input"> Input </label>
+		<input
+			id="sample-form-input"
+			name="title"
+			class="input"
+			type="text"
+			placeholder="e.g., Why is the sky blue?"
+		/>
+	</div>
+</div>
+<div class="field">
+	<div class="field-section">
+		<label class="field-section-label field-section-label-lg label" for="sample-form-input-lg">
+			Input
+		</label>
+		<input
+			id="sample-form-input-lg"
+			name="title"
+			class="input"
+			type="text"
+			placeholder="e.g., Why is the sky blue?"
+		/>
+	</div>
+</div>
+<div class="field display-flex">
+	<div class="field-section">
+		<button class="button button-primary margin-right-xxs">Submit</button>
+		<button class="button button-danger">Cancel</button>
+	</div>
+</div>
+```
+
+### Modifiers
+
+`.field-addons` is added to attach controls together.
+
+```html
+<div class="field field-addons">
+	<div class="field-section">
+		<div class="select">
+			<select>
+				<option>$</option>
+				<option>€</option>
+			</select>
+		</div>
+	</div>
+	<div class="field-section">
+		<input class="input" type="text" placeholder="Transfer money" />
+	</div>
+	<div class="field-section">
+		<a class="button button-primary"> Transfer </a>
+	</div>
+</div>
+```
+
+## Form Section
+
+Form control elements are wrapped with a `.field-section` container.
+
+`.field-section-message` is placed after the element for form validation messages.
+
+```html
+<div class="field">
+	<div class="field-section">
+		<label class="field-section-label field-section-label-lg label" for="sample-form-input-note">
+			Input
+		</label>
+		<input
+			id="sample-form-input-note"
+			name="title"
+			class="input"
+			type="text"
+			placeholder="e.g., Why is the sky blue?"
+			aria-describedby="section-note"
+		/>
+		<p id="section-note" class="field-section-message help help-danger">Sample warning message.</p>
+	</div>
+</div>
+```
+
+### Modifiers
+
+`.field-section-icons` is used when the control element contains icons.
+
+```html
+<div class="field">
+	<div class="field-section field-section-icons">
+		<input class="input input-sm" placeholder="www.microsoft.com" />
+		<span class="icon border" aria-hidden="true"> </span>
+	</div>
+	<div class="field-section field-section-icons margin-top-xxs">
+		<div class="select select-sm">
+			<select>
+				<option>A</option>
+				<option>B</option>
+			</select>
+			<span class="icon border" aria-hidden="true"> </span>
+		</div>
+	</div>
+</div>
+```
+
+`.field-section-icon-right` is used to display the icon on the right side of the element.
+
+```html
+<div class="field">
+	<div class="field-section field-section-icons field-section-icon-right">
+		<input class="input input-sm" placeholder="www.microsoft.com" />
+		<span class="icon border" aria-hidden="true"> </span>
+	</div>
+</div>
+```
```

Comment:
Related to the comment above about reducing the markup to field, field-label, field-body:

Can we decouple input groups / input addons from "field". That seems like a separate thing.

Reply:
There's only one use case of addon in Docs and it seems to be misused, Perhaps it can just be removed.

Reply:
ok let's remove it here, I don't think input addons is part of the form field markup / css.

Reply:
Removed